### ISSUE TITLE
Experiments: Change the way variants are read/written

### ISSUE THREAD

**Issue by nollymar** [dotCMS/core#24964]
### Parent Issue

https://github.com/dotCMS/core/issues/22222

### Problem Statement

The logic in `ESContentFactoryImpl#findAllVersions` was modified to call the method `ESContentFactoryImpl#getContantletInodesFromJsonField`, but the thing is that it fails to return all contentlet versions if the `contentlet_as_json` column is null. For this reason, the `ESContentFactoryImpl#getContantletInodesFromJsonField`should be refactored to avoid using the `contentlet_as_json` column to get variants. The suggested changes are: 
- [x] Create a new column in the `contentlet` table to store/read the `variant` field
- [x] Set a default variant for this new column to avoid `null`values
- [x] Rename get**Contantlet**InodesFromJsonField as it has a typo
- [x] Remove `variantId` from com/dotcms/content/model/Contentlet.java

This change implies the creation of an upgrade task and the modification of the postgres.sql file.

### Steps to Reproduce

This issues came up when testing this card https://github.com/dotCMS/core/issues/24093#issuecomment-1544611454

Basically, when a user comes from an upgrade and tries to list the versions of an existing piece of content (tab on the Edit Content functionality), it should fail to show them all.

### Acceptance Criteria

- [ ] Tests must be implemented for the upgrade task
- [ ] Contentlet versions should be listed correctly when the `contentlet_as_json` and/or `variant` columns are set or not

### dotCMS Version

23.06

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

**Comment by fabrizzio-dotCMS** [comment]
Passing Internal QA
I was able to verify that upon a variant page creation
the new column gets populated  and not the contentlet_as_json 
![Screenshot 2023-06-05 at 11 16 59](https://github.com/dotCMS/core/assets/37148755/7f164f50-9f88-4458-a24d-9ea4111c004c)

However, while testing  the upgrade task   'Task230523CreateVariantFieldInContentlet' ` 
using a large database with 10.000.000 pieces of content. I waited for 1hr but it never completes nor shows any progress 
 
![Screenshot 2023-06-05 at 11 27 22](https://github.com/dotCMS/core/assets/37148755/5e07295c-0d9d-484c-8414-2faeb2f11f30)

**Comment by bryanboza** [comment]
Fixed, testes as part of the card #24093
